src/node/config.ts

```diff
@@ -80,6 +85,14 @@ export interface SharedConfig {
    * https://github.com/vuejs/vue-next/blob/master/packages/compiler-core/src/options.ts
    */
   vueCompilerOptions?: CompilerOptions
+  /**
+   * Customer template compiler for global sfc template block.
+   */
+  compiler?: TemplateCompiler
```

Comment:
Maybe prefix these with `vue` to stay consistent, since vite is not just for Vue?

Reply:
You may or may not need to add https://github.com/RobbinBaauw/vite/blob/9f76f26eafb1267f25485fac5e52b856627a789b/src/node/config.ts#L450 as well